pkg/subctl/cmd/gather/gather.go

```diff
@@ -246,33 +233,27 @@ func gatherBroker(dataType string, info Info) bool {
 	case Resources:
 		brokerRestConfig, brokerNamespace, err := restconfig.ForBroker(info.Submariner, info.ServiceDiscovery)
 		if err != nil {
-			info.Status.QueueFailureMessage(fmt.Sprintf("Error getting the broker's rest config: %s", err))
+			info.Status.Failure("Error getting the broker's rest config: %s", err)
 			return true
 		}
 
 		if brokerRestConfig != nil {
 			info.RestConfig = brokerRestConfig
 
-			info.DynClient, info.ClientSet, err = restconfig.Clients(brokerRestConfig)
-			if err != nil {
-				info.Status.QueueFailureMessage(fmt.Sprintf("Error getting the broker client: %s", err))
-				return true
-			}
-		} else {
```

Comment:
I think that dropping this `else` is what’s causing your problems retrieving `Broker` objects.

Reply:
yes, I missed that the clients are created by using different restconfigs in the if-else loop.